File: golden-paths/frontend/astro/test/contract.test.ts
// Contract self-test (Astro). jest cannot transform .astro components, so the lane tests the shared
// greeting module + the endpoint handlers directly (they are plain functions returning a Response).
import { greeting, SERVICE_NAME } from '../src/lib/greeting';
import { GET as health } from '../src/pages/health';
import { GET as ready } from '../src/pages/ready';
import { GET as hello } from '../src/pages/hello';
import { GET as metrics } from '../src/pages/metrics';

test('greeting is the known payload', () => {
  expect(greeting()).toEqual({ service: SERVICE_NAME, message: 'hello, weyland' });
});

test('GET /health', async () => {
  const res = health();
  expect(res.status).toBe(200);
  expect(await res.json()).toEqual({ status: 'ok' });
});

test('GET /ready', async () => {
  expect(await ready().json()).toEqual({ status: 'ready' });
});

test('GET /hello returns the known payload', async () => {
  const res = hello();
  expect(await res.json()).toEqual({ service: SERVICE_NAME, message: 'hello, weyland' });
});

test('GET /metrics exposes prometheus', async () => {
  hello();
  const body = await (await metrics()).text();
  expect(body).toContain('golden_hello_requests_total');
});

File: golden-paths/dart/shelf/test/contract_test.dart
// Golden-path self-test (Dart/shelf) — the lane probe + contract proof.
// Drives the handler directly (no live socket), mirroring the go exemplar's httptest approach.
import 'dart:convert';

import 'package:shelf/shelf.dart';
import 'package:test/test.dart';

import 'package:golden_dart_shelf/router.dart';

Future<Response> _get(String path) {
  final handler = buildRouter();
  return Future.value(handler(Request('GET', Uri.parse('http://localhost:8080$path'))));
}

void main() {
  test('GET /health returns 200 + status ok', () async {
    final resp = await _get('/health');
    expect(resp.statusCode, 200);
    final body = jsonDecode(await resp.readAsString()) as Map<String, dynamic>;
    expect(body['status'], 'ok');
  });

  test('GET /ready returns 200 + status ready', () async {
    final resp = await _get('/ready');
    expect(resp.statusCode, 200);
    final body = jsonDecode(await resp.readAsString()) as Map<String, dynamic>;
    expect(body['status'], 'ready');
  });

  test('GET /hello returns the known payload', () async {
    final resp = await _get('/hello');
    expect(resp.statusCode, 200);
    final body = jsonDecode(await resp.readAsString()) as Map<String, dynamic>;
    expect(body['service'], serviceName);
    expect(body['message'], 'hello, weyland');
  });

  test('GET /metrics exposes Prometheus text', () async {
    await _get('/hello');
    final resp = await _get('/metrics');
    expect(resp.statusCode, 200);
    final text = await resp.readAsString();
    expect(text, contains('golden_hello_requests_total'));
  });
}
